database/src/migrations/20230823000000_survey_partnerships.ts

```diff
@@ -0,0 +1,231 @@
+  COMMENT ON COLUMN survey_first_nation_partnership.update_user IS 'The id of the user who updated the record as identified in the system user table.'
+  ;
+  COMMENT ON COLUMN survey_first_nation_partnership.revision_count IS 'Revision count used for concurrency control.'
+  ;
+  COMMENT ON TABLE survey_first_nation_partnership IS 'A associative entity that joins projects and first nations.'
+  ;
+
+  ----------------------------------------------------------------------------------------
+  -- Create new keys and indices
+  ----------------------------------------------------------------------------------------
+
+  -- Add foreign key constraint from child table to parent table on survey_id
+  ALTER TABLE survey_stakeholder_partnership ADD CONSTRAINT survey_stakeholder_partnership_fk1
+    FOREIGN KEY (survey_id)
+    REFERENCES survey(survey_id);
+
+  ALTER TABLE survey_first_nation_partnership ADD CONSTRAINT survey_first_nation_partnership_fk1
+    FOREIGN KEY (survey_id)
+    REFERENCES survey(survey_id);
+
+  -- Add foreign key index
+  CREATE INDEX survey_stakeholder_partnership_idx1 ON survey_stakeholder_partnership(survey_id);
+
+  CREATE INDEX survey_first_nation_partnership_idx1 ON survey_first_nation_partnership(survey_id);
+
+  -- Add unique constraint
+  CREATE UNIQUE INDEX survey_stakeholder_partnership_uk1 ON survey_stakeholder_partnership(name, survey_id);
+
+  CREATE UNIQUE INDEX survey_first_nation_partnership_uk1 ON survey_first_nation_partnership(first_nations_id, survey_id);
+
+  -- Create audit and journal triggers
+  create trigger audit_observation before insert or update or delete on survey_stakeholder_partnership for each row execute procedure tr_audit_trigger();
+  create trigger journal_observation after insert or update or delete on survey_stakeholder_partnership for each row execute procedure tr_journal_trigger();
+  create trigger audit_observation before insert or update or delete on survey_first_nation_partnership for each row execute procedure tr_audit_trigger();
+  create trigger journal_observation after insert or update or delete on survey_first_nation_partnership for each row execute procedure tr_journal_trigger();
+  
+  ----------------------------------------------------------------------------------------
+  -- Create views
+  ----------------------------------------------------------------------------------------
+
+  set search_path=biohub_dapi_v1;
+
+  create or replace view survey_stakeholder_partnership as select * from biohub.survey_stakeholder_partnership;
+  create or replace view survey_first_nation_partnership as select * from biohub.survey_first_nation_partnership;
+
+ 
+  ----------------------------------------------------------------------------------------
+  -- Update api_delete_survey procedure
+  ----------------------------------------------------------------------------------------
+
+
+  set search_path=biohub;
+
+  CREATE OR REPLACE PROCEDURE api_delete_survey(p_survey_id integer)
```

Comment:
We probably also need to update the project delete function,as it will still be referencing the old partnerships tables

Reply:
Good catch! Done.